components/wallet/lnc.js

```diff
@@ -0,0 +1,190 @@
+  badges: ['send only', 'non-custodialish', 'budgetable']
+}
+
+const XXX_DEFAULT_PASSWORD = 'password'
+
+export async function validate ({ me, logger, pairingPhrase, password }) {
+  const lnc = await getLNC({ me })
+  try {
+    lnc.credentials.pairingPhrase = pairingPhrase
+    logger.info('connecting ...')
+    // FIXME: this fails with this error:
+    //   Cannot assign to read only property 'undefined' of object '#<Window>'
+    await lnc.connect()
+    logger.ok('connected')
+    logger.info('validating permissions ...')
+    await validateNarrowPerms(lnc)
+    logger.ok('permissions ok')
+    lnc.credentials.password = password || XXX_DEFAULT_PASSWORD
+    logger.info('getting lightning info ...')
+    await lnc.lightning.getInfo()
+    logger.ok('info received')
+  } finally {
+    // FIXME: this fails with this error:
+    //   Cannot read properties of undefined (reading 'wasmClientDisconnect')
+    // uncommented because it shadows the error from lnc.connect()
+    // lnc.disconnect()
+  }
+}
+
+export const schema = object({
+  pairingPhrase: array()
+    .transform(function (value, originalValue) {
+      if (this.isType(value) && value !== null) {
+        return value
+      }
+      return originalValue ? originalValue.split(/[\s]+/) : []
+    })
+    .of(string().trim().oneOf(bip39Words, ({ value }) => `'${value}' is not a valid pairing phrase word`))
+    .min(2, 'needs at least two words')
+    .max(10, 'max 10 words')
+    .required('required'),
+  password: string()
+})
+
+const mutex = new Mutex()
+
+async function unlock ({ lnc, password, status, showModal, logger }) {
+  if (status === Status.Enabled) return password
+
+  return await new Promise((resolve, reject) => {
+    const cancelAndReject = async () => {
+      reject(new Error('password canceled'))
+    }
+    showModal(onClose => {
+      return (
+        <Form
+          initial={{
+            password: ''
+          }}
+          onSubmit={async (values) => {
+            try {
+              lnc.credentials.password = values?.password
+              logger.ok('wallet enabled')
+              onClose()
+              resolve(values.password)
+            } catch (err) {
+              logger.error('failed attempt to unlock wallet', err)
+              throw err
+            }
+          }}
+        >
+          <h4 className='text-center mb-3'>Unlock LNC</h4>
+          <PasswordInput
+            label='password'
+            name='password'
+          />
+          <div className='mt-5 d-flex justify-content-between'>
+            <CancelButton onClick={() => { onClose(); cancelAndReject() }} />
+            <SubmitButton variant='primary'>unlock</SubmitButton>
+          </div>
+        </Form>
+      )
+    }
+    )
+  })
+}
+
+// FIXME: pass me, status, showModal in useWallet hook
+export async function sendPayment ({ bolt11, pairingPhrase, password: configuredPassword, me, status, showModal, logger }) {
```

Comment:
consider breaking the config, payment request, and context into independent parameters. Maybe something like:

`sendPayment(bolt11: string, config: Object, context: Object)`